modules/ballerina-core/src/main/java/org/ballerinalang/util/semantics/SemanticAnalyzer.java

```diff
@@ -777,10 +777,14 @@ public void visit(BallerinaAction action) {
         }
 
         // First parameter should be of type connector in which these actions are defined.
+        if (action.getParameterDefs().length < 1) {
+            BLangExceptionHelper.throwSemanticError(action, SemanticErrors.INVALID_ACTION_FIRST_PARAMETER,
+                    action.getConnectorDef());
+        }
         ParameterDef firstParamDef = action.getParameterDefs()[0];
         if (firstParamDef.getType() != action.getConnectorDef()) {
```

Comment:
BTW shall we merge these two conditions? And also I think it's better if you can get the error and throw it here instead of throwing it inside BLangExceptionHelper